catalog/controller/extension/module/mobbex_finance_widget.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+require_once DIR_SYSTEM . 'library/mobbex/config.php';
+require_once DIR_SYSTEM . 'library/mobbex/sdk.php';
+
+class ControllerExtensionModuleMobbexFinanceWidget extends Controller
+{
+    /** @var MobbexHelper */
+    public static $helper;
+
+    public function index()
+    {
+        // load models
+		$this->load->model('setting/setting');
+		$this->mobbexConfig = new MobbexConfig($this->model_setting_setting);
+
+        if(
+            (!$this->mobbexConfig->active_product && $this->request->get['route'] == 'product/product') 
+            || (!$this->mobbexConfig->active_cart && $this->request->get['route'] == 'checkout/cart')
+            || !$this->mobbexConfig->status_widget
+        ) {
+            return;
+        }
+
+		//Init sdk classes
+		\MobbexSdk::init($this->mobbexConfig);
+
+        $data = [
+            'price'   => $this->getPrice($this->request->get['route']),
+            'sources' => \Mobbex\Repository::getSources($this->getPrice($this->request->get['route'])),
+			'theme'   => 'light',
+            'button'  => [
+                'custom_styles' => $this->mobbexConfig->button_styles,
+                'text'          => $this->mobbexConfig->button_text,
+                'logo'          => $this->mobbexConfig->button_logo,
+            ]
+        ];
+
+        // Return view
+        return $this->load->view('extension/mobbex/finance_widget', $data);
+    }
+
+    /**
+     * Calculates the price to show in widget.
+     * 
+     * @param string $route
+     * 
+     * @return float
+     */
+    public function getPrice($route)
```

Comment:
Acá recibís la ruta como param, pero para encontrarla hacés uso de `$this->request->get`, que ya se usa dentro de este método. Entonces, si la idea es aislar la lógica para que los datos que vienen por request tengan que venir si o si por param también habría que pasar el `product_id` (cosa que no me parece lo mejor). Para mí lo ideal sería que no reciba ningún param y que esté explicado en la doc del método que devuelve el precio dependiendo de la ruta actual.